specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-05-31/consumption.json

```diff
@@ -1324,148 +2066,653 @@
-            "name": "ChargeType",
-            "modelAsString": true
+          "name": "OperatorType",
+          "modelAsString": true
           }
         },
-        "confidenceLevels": {
-          "description": "The details about the forecast confidence levels. This is populated only when chargeType is Forecast.",
+        "values": {
+          "description": "Array of values to use for comparison",
           "type": "array",
           "items": {
-            "type": "object",
-            "properties": {
-              "percentage": {
-                "description": "The percentage level of the confidence",
-                "type": "number",
-                "format": "decimal",
-                "readOnly": true
-              },
-              "bound": {
-                "description": "The boundary of the percentage, values could be 'Upper' or 'Lower'",
-                "type": "string",
-                "enum": [
-                  "Upper",
-                  "Lower"
-                ],
-                "x-ms-enum": {
-                  "name": "Bound",
-                  "modelAsString": true
-                }
-              },
-              "value": {
-                "description": "The amount of forecast within the percentage level",
-                "type": "number",
-                "format": "decimal",
-                "readOnly": true
-              }
-            }
+            "type": "string"
           },
-
-          "readOnly": true
+          "minItems": 1
+        }
+      },
+      "required": [
+        "name",
+        "operator",
+        "values"
+      ]
+    },
+    "DimensionsListResult": {
+      "description": "Result of listing dimensions. It contains a list of available dimensions.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of dimensions.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Dimension"
+          }
+        }
+      }
+    },
+    "Dimension": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DimensionProperties",
+          "title": "Dimension properties"
+        }
+      }
+    },
+    "DimensionProperties": {
+      "type": "object",
+      "properties": {
+        "data": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "total": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "category": {
+          "type": "string"
+        },
+        "usageStart": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "usageEnd": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "nextLink": {
```

Comment:
Why would this object get a nextLink? Just checking that it's accurate, since it's not usual and SDK want be able to do anything with that, and polling would be manual.

Reply:
Hi @imazuel 
This is intentional. The response here is a single object and not a list of object. The single object however, can have a very big payload in data field. Hence, the need for pagination. We are aware that SDKs will not be able to parse it automatically and we will have additional work to support this in SDK. 
This pattern is already used in one other API v.i.z. Pricesheets. We had multiple back and forth discussions with ARM team and this is the only way we can come up with since the response here is a single ARM resource object and not an array of ARM resources.